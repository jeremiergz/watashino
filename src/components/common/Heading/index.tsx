import styled from 'styled-components';
import { variant } from 'styled-system';
import Text, { TextProps } from '../../primitives/Text';

const Heading = styled(Text)<HeadingProps>`
  display: inline-block;
  ${variant({
    variants: {
      h1: { fontSize: 44, fontWeight: 'bold' },
      h2: { fontSize: 32 },
      h3: { fontSize: 24 },
      h4: { fontSize: 20 },
    },
  })}
`;

Heading.displayName = 'Heading';

export type HeadingProps = TextProps & {
  variant: 'h1' | 'h2' | 'h3' | 'h4';
};
export default Heading;
