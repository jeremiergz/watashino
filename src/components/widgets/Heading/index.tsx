import React from 'react';
import styled from 'styled-components';
import { variant } from 'styled-system';
import theme from '../../../theme';
import Text, { TextProps } from '../../primitives/Text';

type HeadingProps = TextProps & {
    variant: 'h1' | 'h2' | 'h3' | 'h4';
};

const Heading = styled(Text)<HeadingProps>`
    ${variant({
        variants: {
            h1: { fontSize: 44, fontWeight: theme.fonts.main.weight.bold },
            h2: { fontSize: 32 },
            h3: { fontSize: 24 },
            h4: { fontSize: 20 },
        },
    })}
`;

const HeadingComponent = ({ children, variant, ...rest }: HeadingProps) => {
    return (
        <Heading as={variant} variant={variant} {...rest}>
            {children}
        </Heading>
    );
};

HeadingComponent.displayName = 'Heading';

export { HeadingProps };
export default HeadingComponent;
