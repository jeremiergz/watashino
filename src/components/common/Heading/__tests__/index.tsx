import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Heading from '../../Heading';

describe('components/common/Heading Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = <Heading variant="h1">Heading Component</Heading>;
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
