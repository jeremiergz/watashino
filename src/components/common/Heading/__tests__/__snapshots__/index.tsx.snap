// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/Heading Test Suite matches snapshot 1`] = `
<h1
  className="sc-AykKC sc-AykKD dnmvVU"
>
  Heading Component
</h1>
`;
